### PR TITLE
feat: Price charts

### LINKED ISSUE #1157: Feature: Token Price chart
## Background
Currently we only reflect real time token prices. We're missing backend function to fetch token price changes over time & a visual chart tool to display it.

## Task
- [x] #1158 
- [x] #1159 

### Price API
API endpoint to fetch token market prices over timestamp: 
```
https://price-api.crypto.com/price/v2/all/{slug}?convert={currency}
```
e.g. https://price-api.crypto.com/price/v2/all/ethereum?convert=USD

### Visual Chart plotting tool
- Determine which visual tool library can be best fitted

### PR DESCRIPTION
**Pull request by XinyuCRO**
Closes #1157 #1158 #1159

image

### CODE REVIEW

File: src/pages/assets/components/ChartArea.tsx [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    getContainerSize();
+    window.addEventListener('resize', getContainerSize);
+    return () => window.removeEventListener('resize', getContainerSize);
+  }, []);
+
+  const fetchTokenPrices = useCallback(async () => {
+    setLoadingData(true);
+    try {
+      const data = await croMarketPriceApi.getTokenPrices(
+        `${asset.symbol}`,
```
Comment (crypto-matto) [inline src/pages/assets/components/ChartArea.tsx:58]:
```suggestion
        `${asset.mainnetSymbol}`,
```
Testnet tokens like TCRO is not fetching any data. Use `asset.mainnetSymbol` instead

**Comment by crypto-matto** [comment]
### Issue
- #967 
```
getTokenPrices(cryptoSymbol: string, fiatCurrency: string, interval: string) {
... 
    const tokenSlugInfo = allTokensSlugMap?.find(tokenSlug => tokenSlug.symbol === cryptoSymbol);
    const slug = tokenSlugInfo?.slug;
... 
}
```
The problem with using barely `cryptoSymbol` is that it may fetch token prices of identical symbol tokens from other chains, resulting in an incorrect price chart. Therefore we have to identify the corresponding slug based on more criteria: 
1. For CRC20 tokens, check whether the API returned data `tags` includes `cronos-ecosystem`
2. For ERC20 tokens, sadly the above method doesn't work as many of them includes `ethereum-ecosystem`. I could only apply our whitelistedERC20Tokens whitelist table for filtering. 

You may reference on `getTokenPriceFromCryptoCom()`, see how I handled it. You may revise it and put the redundant logic into another function.

**Comment by XinyuCRO** [comment]
if only they respond token list with contract addresses, the token slugs may be the same under the same chain

**Comment by crypto-matto** [comment]
> if only they respond token list with contract addresses, the token slugs may be the same under the same chain

Ya sadly they don't. They don't provide the token contract address from any of their returned data neither. `Slug` is the only thing we got to locate the corresponding token.

**Comment by crypto-matto** [comment]
## Updated
- Added Dynamic price figure
螢幕截圖 2022-08-04 下午5 35 49


- Default Round price figure to 2 decimals

**Review by crypto-matto — DISMISSED** [review]

**Review by crypto-matto — APPROVED** [review]

**Pull request merged**